client/client_test.go

```diff
@@ -45,8 +44,24 @@ func TestNew(t *testing.T) {
 
 	messageserviceA.Connect(messageserviceB)
 	messageserviceB.Connect(messageserviceA)
-	clientA.CreateDirectChannel(b, types.Address{}, types.Bytes{}, outcome.Exit{}, big.NewInt(0))
+	// Set up an outcome that requires both participants to deposit
+	outcome := outcome.Exit{outcome.SingleAssetExit{
+		Allocations: outcome.Allocations{
+			outcome.Allocation{
+				Destination: types.AddressToDestination(a),
+				Amount:      big.NewInt(5),
+			},
+			outcome.Allocation{
+				Destination: types.AddressToDestination(b),
+				Amount:      big.NewInt(5),
+			},
+		},
+	}}
 
-	<-time.After(time.Second)
+	id, res := clientA.CreateDirectChannel(b, types.Address{}, types.Bytes{}, outcome, big.NewInt(0))
+	got := <-res
 
+	if got.ObjectiveId != id {
+		t.Errorf("expected completed objective with id %v, but got %v", id, got.ObjectiveId)
+	}
```

Comment:
This is great and will close #205 !

Reply:
Thinking about it, it would be even better to assert on *both* clients getting an objective completed signal. Of course clientB does not have a "per channel" chan to listen on, because their objective was spawned via a message, not an API call. So probably they need to listen through one of the other pipes?

Reply:
I've updated the test to now wait for client B's complete signal in [b7a69f6](https://github.com/statechannels/go-nitro/pull/208/commits/b7a69f6d86d03fb1394fb827362cd13ef0eb2242). Since we expose a channel off the client for **any** completed objective, we simply use that for client B.